pkg/client/testing/core/fixture.go

```diff
@@ -18,28 +18,37 @@ package core
 
 import (
 	"fmt"
-	"io/ioutil"
 	"reflect"
 	"strings"
+	"sync"
 
 	"k8s.io/kubernetes/pkg/api/errors"
 	"k8s.io/kubernetes/pkg/api/meta"
 	"k8s.io/kubernetes/pkg/api/unversioned"
 	"k8s.io/kubernetes/pkg/client/restclient"
 	"k8s.io/kubernetes/pkg/runtime"
-	"k8s.io/kubernetes/pkg/util/yaml"
 	"k8s.io/kubernetes/pkg/watch"
 )
 
-// ObjectRetriever abstracts the implementation for retrieving or setting generic
-// objects. It is intended to be used to fake calls to a server by returning
-// objects based on their kind and name.
-type ObjectRetriever interface {
-	// Kind should return a resource or a list of resources (depending on the provided kind and
-	// name). It should return an error if the caller should communicate an error to the server.
-	Kind(gvk unversioned.GroupVersionKind, name string) (runtime.Object, error)
-	// Add adds a runtime object for test purposes into this object.
+// Registry keeps track of objects. It is intended to be used to fake
+// calls to a server by returning objects based on their kind,
+// namespace and name.
+type Registry interface {
+	// Get retrieves the object by its kind, namespace and name. If
+	// kind is a list, all objects that match ns and name are returned
+	// in their corresponding list enclosure (e.g. PodList,
+	// ServiceList).
+	Get(gvk unversioned.GroupVersionKind, ns, name string) (runtime.Object, error)
```

Comment:
Yeah, it will make things cleaner for sure. I am still not clear on a distinction between GroupResource and GroupKind: existing code seems to be using them interchangeably.